tf_agents/utils/batched_observer_unbatching.py

```diff
@@ -0,0 +1,51 @@
+# coding=utf-8
+# Copyright 2022 The TF-Agents Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Adapter for using unbatched observers in batched contexts."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from typing import Callable
+
+from tf_agents.trajectories import trajectory as trajectory_lib
+from tf_agents.utils import nest_utils
+
+Observer = Callable[[trajectory_lib.Trajectory], None]
```

Comment:
This is only used once. You don't have to create a variable for it.

Reply:
I created that for readability since:
```py
Callable[[], Observer]
```
seems to me to read better than:
```py
Callable[[], Callable[[trajectory_lib.Trajectory], None]]
```
WDYT?

Reply:
Callable[[], Callable[[trajectory_lib.Trajectory], None]] is fine, so people don't have to check what Observer means to use this function. If you want to use this in multiple places or multiple files, it's better to create a variable for it. If you want to elaborate on this, you can mention it in the comments section of __init__

Reply:
Done.